src/SQLStore/SimpleListLookup.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace SMW\SQLStore;
+
+/**
+ * A basic interface for fetching a simple list either from a DB or cache
+ * instance
+ *
+ * @license GNU GPL v2+
+ * @since 2.2
+ *
+ * @author mwjames
+ */
+interface SimpleListLookup {
```

Comment:
> If SimpleListLookup has only the fetchResultList method then one has always to check that it is SimpleListLookup or a concrete ListLookupCache instance because isCached is getTimestamp is only provided by ListLookupCache.

What makes it so that users of this thing inherently need to be able to interact with the cache stuff? If you can avoid this, then what I suggest works. And I do not see why you'd not be able to avoid it.